pgmpy/models/BayesianModel.py

```diff
@@ -442,6 +442,12 @@ def get_independencies(self, latex=False):
                     if independent_variables:
                         independencies.add_assertions([start, independent_variables,
                                                        observed])
+            independencies_list = independencies.get_assertions()
+            local_independencies_list = self.local_independencies(start)
+            assertions_list = local_independencies_list.get_assertions()
+            for assertion in assertions_list:
```

Comment:
@kshitij10496 We won't need to do all this if `self.independencies` in `Independencies` class is a `set`. So, I would suggest you to change that also. And then you will also need to add a `__hash__` method in `IndependenceAssertion` class so that you can put it in a `set`.